File: MyeongheonHong/2023년/12월/12월 1주차/13305.js
// URL : https://www.acmicpc.net/problem/13305

const fs = require('fs');
const filePath =
  process.platform === 'linux' ? '/dev/stdin' : '../../input.txt';
let input = fs.readFileSync(filePath).toString().trim();
input = input.split('\n');
const INPUT_LENGTH = Number(input.shift());

const distance = input[0].split(' ').map((e) => BigInt(e));
const price = input[1].split(' ').map((e) => BigInt(e));

function solution() {
  let length = distance.reduce((acc, curr) => BigInt(acc) + curr, 0);
  let answer = price[0] * length;
  let sum = 0n;

  for (let i = 0; i < price.length - 1; i++) {
    answer = Math.min(String(answer), String(sum + length * price[i]));
    length -= distance[i];
    sum += price[i] * distance[i];
  }
  console.log(String(answer));
}

solution();

File: SungjunChoi/24-01-23/13305.py
n=int(input())
r=[]
s=input()
t=s.split()
r.append(0)
for i in t: r.append(int(i))
p=[]
s=input()
t=s.split()
for i in t: p.append(int(i))
l=[]
k=[]
last=p[0]
result=0
for i in range(len(p)):
 if p[i]<=last:
  l.append(i)
  last=p[i]
last=0
for i in r:
 k.append(last+i)
 last+=i
last=0
for i in range(len(l)-1):
 t=l[i]
 t2=l[i+1]
 result+=(k[t2]-last)*p[t]
 last=k[t2]
if last<k[len(k)-1]: result+=(k[len(k)-1]-last)*p[len(l)-1]
print(result)
